src/pkcs11/core_pkcs11_test.c

```diff
@@ -0,0 +1,2647 @@
+/*
+ * FreeRTOS FreeRTOS LTS Qualification Tests preview
+ * Copyright (C) 2022 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_pkcs11_test.c
+ * @brief Integration tests for the corePKCS11 implementation.
+ */
+/* Standard includes. */
+#include <stdlib.h>
+#include <string.h>
+
+/* corePKCS11 includes. */
+#include "core_pki_utils.h"
+#include "core_pkcs11.h"
+
+/* Developer mode provisioning. */
+#include "dev_mode_key_provisioning.h"
+
+/* Test includes. */
+#include "unity_fixture.h"
+#include "unity.h"
+
+/* mbedTLS includes. */
+#include "mbedtls/sha256.h"
+#include "mbedtls/pk.h"
+#include "mbedtls/pk_internal.h"
+#include "mbedtls/oid.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/entropy_poll.h"
+#include "mbedtls/x509_crt.h"
+
+/* corePKCS11 test includes. */
+#include "core_pkcs11_test.h"
+#include "rsa_test_credentials.h"
+#include "ecdsa_test_credentials.h"
+
+/* Test configuration includes. */
+#include "test_execution_config.h"
+#include "test_param_config.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The index of the slot that should be used to open sessions for PKCS #11 tests.
+ *
+ * corePKCS11 utilities are also used in this test. Currenlty, the slot number has to be 0.
+ */
+#define PKCS11_TEST_SLOT_NUMBER    ( 0 )
+
+/**
+ * @brief The PKCS #11 label for the object to be used for code verification.
+ *
+ * Used by over-the-air update code to verify an incoming signed image.
+ *
+ * For devices with on-chip storage, this should match the non-test label.
+ * For devices with secure elements or hardware limitations, this may be defined
+ * to a different label to preserve AWS IoT credentials for other test suites.

```

Comment:
JITP certificate supported is claimed in core_pkcs11_config.h with pkcs11configJITP_CODEVERIFY_ROOT_CERT_SUPPORTED config. The corePKCS11 test also use this macro to enable create/destory JITP objects. Since other test should not rely on the result of this test cases, I suggest we remove the comments here.